zebra-state/src/service.rs

```diff
@@ -664,6 +664,17 @@ impl Service<Request> for StateService {
                     .collect();
                 async move { Ok(Response::BlockHeaders(res)) }.boxed()
             }
+            Request::IsLegacyChain => {
+                let mut response = None;
+                if let Some(tip) = self.best_tip() {
+                    if legacy_chain_check(tip.0, self.any_ancestor_blocks(tip.1), self.network)
+                        .is_err()
+                    {
+                        response = Some(self.disk.path().to_path_buf());
+                    }
+                }
+                async move { Ok(response).map(Response::LegacyChain) }.boxed()
```

Comment:
Minor: this could be done without the `.map`. I'm not sure which one is easier to read, so feel free to leave it as is :D 

```suggestion
                async move { Ok(Response::LegacyChain(response)) }.boxed()
```

I'm also thinking if it might make sense to rename `response` into something like `legacy_chain_path` :thinking:

Reply:
done at https://github.com/ZcashFoundation/zebra/pull/2366/commits/0587bed76c3d302a01e4a20ed943c2ba11b79214